tests/lax_numpy_test.py

```diff
@@ -200,6 +200,18 @@ def op_record(name, nargs, dtypes, shapes, rng_factory, diff_modes,
               inexact=True, tolerance={np.float64: 1e-9}),
 ]
 
+def _logaddexp_override(x1, x2):
+  if np.issubdtype(x1.dtype, np.complexfloating) or np.issubdtype(x2.dtype, np.complexfloating):
+    return np.log(np.exp(x1) + np.exp(x2))
+  else:
+    return np.logaddexp(x1, x2)
+
+def _logaddexp2_override(x1, x2):
+  if np.issubdtype(x1.dtype, np.complexfloating) or np.issubdtype(x2.dtype, np.complexfloating):
+    return np.log2(np.exp2(x1) + np.exp2(x2))
+  else:
+    return np.logaddexp2(x1, x2)
```

Comment:
You can define these functions within tests of logaddexp and logaddexp2.

For and example of this, here is a test of functionality offered by `jnp.nonzero` that does not have an equivalent in `numpy`: https://github.com/google/jax/blob/cbfa539053d7cab3f07309504fb6bb5454dd91f2/tests/lax_numpy_test.py#L946-L952

Rather than trying to greate a general method to override numpy functions, you can just write a specific check for the behavior you want to check.